src/server/report/report.service.ts

```diff
@@ -0,0 +1,107 @@
+import { Injectable, Inject } from '@nestjs/common';
+import { Stream } from 'stream';
+import Excel from 'exceljs';
+import { isSEASEnumToString, offeredEnumToString } from 'common/constants';
+import { CourseInstanceService } from '../courseInstance/courseInstance.service';
+import CourseInstanceResponseDTO from '../../common/dto/courses/CourseInstanceResponse';
+import { formatMeetingForReport, MULTILINE_SEPARATOR } from './utils';
+
+@Injectable()
+export class ReportService {
```

Comment:
There is also going to be a Faculty Report, but the idea is that it would be generated by a separate method, so you'd have `ReportService#streamCourseReport` and `ReportService#streamFacultyReport`, each with corresponding methods under `ReportController` that would handle the routes `/report/courses` and `/report/faculty`.

Reply:
I did wonder about that. I guess what I was wondering was if it made sense to have those two reports in one service/class like that, or if they're different enough that it would make sense to split them into different classes...

Reply:
I don't see any reason to split them. I think it makes more sense to keep the two reports together. Even though they're acting on different entities, I think they're more similar to each other in terms of response structure and formatting than either is to its entity-sibling methods (i.e. the other `CourseService` methods).

Keeping them in the same Service also keeps all the Excel-generating logic in one place, so from a developer experience standpoint it's easier to reference back to existing code when adding additional reports, etc.

Reply:
That's true....